### PR TITLE
fix: improved error message on broken CIDv0

### PR DESCRIPTION
**Pull request by rvagg**
move @lidel's fix from https://github.com/ipfs/go-cid/pull/116

### CODE REVIEW

File: path.go [hunk 1]
```diff
@@ -172,11 +172,19 @@ func SplitAbsPath(fpath Path) (cid.Cid, []string, error) {
...
 		return cid.Cid{}, nil, &pathError{error: fmt.Errorf("invalid CID: %s", err), path: string(fpath)}
 	}
 
 	return c, parts[1:], nil
 }
+
+func decodeCid(cstr string) (cid.Cid, error) {
+	c, err := cid.Decode(cstr)
+	if err != nil && len(cstr) == 46 && cstr[:2] == "qm" { // https://github.com/ipfs/go-ipfs/issues/7792
+		return cid.Cid{}, fmt.Errorf("%v: This looks like a CIDv0 that has been lowercased. Convert the CIDv0 to CIDv1 in case-insensitive base32 using 'ipfs cid base32 <Qm..>' and try again.", err)
```
Comment (Stebalien) [inline path.go:187]: I would still like this to be much shorter. Can we just say "this looks like a lowercase CID"? Telling the user what to do is a bit out of scope for an error.

**Review by Stebalien — COMMENTED** [review]
cc @lidel

File: path.go (hunk 1 above)
Comment (jessicaschilling) [reply path.go:187]: 'This looks like a CIDv0 that has been lowercased; you may wish to convert to case-agnostic CIDv1' or similar? Intent would be not to tell user what to do, but hint that there's a workaround if preserving case isn't an option.

**Comment by Stebalien** [comment]
That works. Could also say "consider converting it to base32 first". We
can also include short links.

**Comment by rvagg** [comment]
pushed a fixup with:

> This looks like a CIDv0 that has been lowercased; consider converting to a case-agnostic CIDv1, such as base32

How's that sound?

**Comment by Stebalien** [comment]
It should start lower case and I'd prefer if it were a bit shorter, but that's not a huge issue.

**Comment by rvagg** [comment]
`"%v (possible lowercased CIDv0; consider converting to a case-agnostic CIDv1, such as base32)"`

how's this for a compromise between size and communicating what's wrong @Stebalien @jessicaschilling?

The construction does get a bit whacky when you look at it in full, which is why I'm trying out the parens:

invalid path "/ipfs/qmbwqxbekc3p8tqskc98xmwnzrzdtrlmimpl8wbutgsmnr": invalid CID: selected encoding not supported (possible lowercased CIDv0; consider converting to a case-agnostic CIDv1, such as base32)

**Comment by jessicaschilling** [comment]
SGTM - it's a little awkward but good enough given the constraints.

**Review by Stebalien — APPROVED** [review]

**Comment by Stebalien** [comment]
LGTM! (although the test needs fixing). I generally avoid long error messages as we tend to glue them together.

**Comment by rvagg** [comment]
Sorry, I hadn't properly pushed the version I had locally which had it with parens and the test working. I've squashed now and force pushed the single commit, the message is `"%v (possible lowercased CIDv0; consider converting to a case-agnostic CIDv1, such as base32)"`.

I'll take this as approved and merge shortly.

**Review by Stebalien — APPROVED** [review]

**Comment by Stebalien** [comment]
too slow

**Comment by lidel** [comment]
@Stebalien @aschmahmann  would it be ok to make a release with this? 
I'd love to sneak this into go-ipfs 0.8.0 (https://github.com/ipfs/go-ipfs/pull/7807)

**Comment by aschmahmann** [comment]
@lidel SGTM

**Pull request merged**